### DISCUSSION TITLE
How to get lazy loading and modify defaults without running in variable not existing ?

### DISCUSSION THREAD

**Discussion by tamwile** [radian-software/apheleia#281] [Q&A] [answered]
When following the user guide, 
`(apheleia-global-mode +1)` seems a simple configuration line.

But my issue arise when i want to modify apheleia-formatters.
Again, i tried to follow the user guide
`(setf (alist-get 'black apheleia-formatters)
      '("black" "--option" "..." "-"))`

But in this case, i obviously get the error: 
"Symbol’s value as variable is void: apheleia-formatters"
I tried to wrap the sexp into a function and to hook the function to apheleia-global-mode-hook, but it doesn't work

I managed to make it work by hooking into apheleia-post-format-hook, but:
1. it only works after the first save (second save and more)
2. it setf apheleia-formatters every time i format the buffer

How to configure apheleia to be able to get lazy loading thanks to apheleia-global-mode and at the same time setting a new default value for apheleia-formatters without running into all these errors and weird hacks ?

**Comment by raxod502** [comment, marked as answer]
You want:

```elisp
(with-eval-after-load 'apheleia-formatters
  (setf (alist-get 'black apheleia-formatters) '("black" "--option" "..." "-")))
```

**Reply by tamwile** [reply]
Truthfully, i expected an example with use-package or leaf, but your answer allowed me to solve my problem.
Thank you !

**Reply by raxod502** [reply]
`use-package` is basically just `with-eval-after-load` plus some convenience. There is really not much else to it :)
